src/tls.c

```diff
@@ -677,12 +678,69 @@ static void updateSSLState(connection *conn_) {
     updatePendingData(conn);
 }
 
+static int getCertFieldByName(X509 *cert, const char *field, char *out, size_t outlen) {
+    if (!cert || !field || !out) return 0;
+
+    int nid = -1;
+
+    if (!strcasecmp(field, "CN"))
+        nid = NID_commonName;
+    else if (!strcasecmp(field, "O"))
+        nid = NID_organizationName;
+    // Add more mappings here as needed
+
+    if (nid == -1) return 0;
+
+    X509_NAME *subject = X509_get_subject_name(cert);
+    if (!subject) return 0;
+
+    return X509_NAME_get_text_by_NID(subject, nid, out, outlen) > 0;
+}
+
+static void autoAuthenticateClientFromCert(tls_connection *conn) {
+    if (!conn || !SSL_is_init_finished(conn->ssl)) return;
+
+    const char *field = server.tls_ctx_config.client_auth_field;
+    if (!field) return;
+
+    serverLog(LL_NOTICE, "TLS: Config Auto Auth Field = %s", field);
+
+    X509 *cert = SSL_get_peer_certificate(conn->ssl);
+    if (!cert) {
+        serverLog(LL_WARNING, "TLS: No peer certificate found after handshake");
+        return;
+    }
+
+    char field_value[256];
+    if (!getCertFieldByName(cert, field, field_value, sizeof(field_value))) {
+        serverLog(LL_WARNING, "TLS: Failed to extract field %s from certificate", field);
+        X509_free(cert);
+        return;
+    }
+
+    serverLog(LL_NOTICE, "TLS: Config Auto Auth %s = %s", field, field_value);
+    conn->c.cert_user = sdsnew(field_value);
+    X509_free(cert);
+
+    client *c = connGetPrivateData(&conn->c);
+    if (!c) return;
+
+    user *u = ACLGetUserByName(conn->c.cert_user, sdslen(conn->c.cert_user));
+    serverLog(LL_NOTICE, "TLS: ACL lookup user: %s", u ? u->name : "NOT FOUND");
+    if (u) {
+        c->user = u;
+        c->flag.authenticated = true;
+        serverLog(LL_NOTICE, "TLS: Auto-authenticated client as %s", u->name);
```

Comment:
Logging twice here is too verbose IMO. Let's log only the second one, when it has been found and authenticated.

Reply:
We also mark u->name as sensitive in other places, we should add it behind the hide user data flag. I would drop the log level to verbose if we want to keep the second one.

Reply:
Done update code